### PR TITLE
Feature/gdb

### PR DESCRIPTION
**Pull request by BaranovMykola**
This classes are core of using gdb in Kuzya. This pull request has class Gdb wich provides basics methods such as step out\into\over, reading variables, etc. Also it has class Variable and Breakpoint. This pull request doesn't make any influence to kuzya, because this classes are not using yet.

### CODE REVIEW

File: src/breakpoint.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#include <QStringList>
+#include <QDebug>
+#include <QRegExp>
+
+void Breakpoint::parse(const QString &line)
+{   // fill breakpoint from $line$
+    /*
+        $"1       breakpoint     keep y   0x0040149e in main() at main.cpp:40\n"$
+        $"2       breakpoint     del y   0x004014fc in main() at main.cpp:41\n"$
+    */
```
Comment (OlyaOlga) [inline src/breakpoint.cpp:12]: Is this comment necessary?

File: src/variable.cpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
+#include "variable.h"
+
+#include <QRegExp>
+#include <QDebug>
+
+Variable::Variable()
+{
+
```
Comment (OlyaOlga) [inline src/variable.cpp]: Delete this line, please.

File: src/variable.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+Variable::Variable(QString name, QString type, QString content):
+    mName(name),
+    mType(type),
+    mContent(content)
+{
+}
+
+QStringList Variable::getSubVariables() const
+{  // returns QStringList consists from nested variable's names (only first level)
+   //  {digit = {real = 3, imaginary = 4}}
```
Comment (OlyaOlga) [inline src/variable.cpp]: Seems like this comment doesn't match the context of the code.

File: src/variable.cpp [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+QStringList Variable::getNestedStructures() const
+{
+    QString nestedType = this->getContent();
+    QStringList res;
+//    do
+//    {
+//        nestedType = this->readNestedStruct(nestedType);
+//        res << nestedType;
+//    }
+//    while(!nestedType.isEmpty());
```
Comment (OlyaOlga) [inline src/variable.cpp]: Delete unused code, please.

File: src/variable.cpp [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        if (behind == " =" && capture)
+        {
+            capture = false;
+        }
+        else if (vec[i] == ',' && level == 1 && !capture || i+1 == vec.size())
+        {
+            capture = true;
+            if(key.isEmpty() && value.isEmpty())
+            {
+                return res;
```
Comment (OlyaOlga) [inline src/variable.cpp]: Why do you need a return statement here?

File: src/variable.cpp [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            }
+            else if(!key.isEmpty() && !value.isEmpty())
+            {
+                value[0] = ' ';
+                value[1] = ' ';
+//                value.append('}');
```
Comment (OlyaOlga) [inline src/variable.cpp]: Is this comment necessary here?

File: src/breakpoint.cpp (hunk 1 above)
Comment (BaranovMykola) [reply src/breakpoint.cpp:12]: This comment shows you format of input string

**Pull request closed without merging**